project_template/BUILDING.md

```diff
@@ -0,0 +1,11 @@
+
+## Requirements
+
+This Azure SDK module has several external build-time requirements.
+
+- **Catch2** (for tests): https://github.com/catchorg/Catch2.git
+
+These may be installed using a package manager or embedded in the external/ folder.

```

Comment:
>external [](start = 66, length = 8)

Consider sub-repo'ing (submodules) all the possible external dependencies. This comes with the great advantage of pinning the used submodule to a specific SHA.

If left to the package manager or user or some other means that will break the build dependency as far as versioning goes.

Reply:
The critical scenarios are:
1. A user should be able to clone, build, and consume *just this library* easily with pinned, tested versions.
2. A user should be able to integrate this library into a larger project which already has copies of `curl`, `libuv`, etc. This is critical to enable the user to resolve security issues by updating dependencies.

Solving (2) implies compatibility with package managers. One way to solve (1) is to submodule all external dependencies. An alternative way to solve (1) is to use an embeddable package manager, such as hunter or vcpkg, though it will obviously be critical to not compromise the "SHA pinning" aspect required by point (1).

Reply:
I actually have support for submodule dependencies. The reason I don't is that I wanted this project to be a subfolder in azure-sdk-for-c, and didn't want to add submodules to the parent repository. I could add a gitmodules file to make them easier to add.

Also @ras0219-msft imo using submodules has the additional advantage that it makes it easier to reduce / fix bugs in your dependencies and submit those fixes upstream.

Another reason I didn't add the submodules is that if the library is building as part of a large build (say the whole azure-sdk-for-c project) then each individual library shouldn't be vendoring anything.